File: hew-core/tests/treesitter/fixtures/sample.go
// Sample Go fixture for TS.4 end-to-end tests.
// Layout: two top-level funcs, one struct + two methods, one closure.

package main

import "fmt"

func AlphaCompute(a, b int) int {
	scale := func(x int) int { return x * 2 }
	return scale(a) + scale(b)
}

func BetaFormat(name string) string {
	return fmt.Sprintf("hello, %s", name)
}

type Widget struct {
	ID int
}

func (w Widget) GammaDescribe() string {
	return fmt.Sprintf("widget-%d", w.ID)
}

func (w Widget) DeltaClone() Widget {
	return Widget{ID: w.ID}
}

func EpsilonDispatch(w Widget) string {
	return w.GammaDescribe()
}

File: hew-core/tests/treesitter/fixtures/sample.rs
// Sample Rust fixture for TS.4 end-to-end tests.
// Layout: two top-level fns, one struct + two methods, one closure.

pub fn alpha_compute(a: i32, b: i32) -> i32 {
    let multiplier = 2;
    let scale = |x: i32| x * multiplier;
    scale(a) + scale(b)
}

pub fn beta_format(name: &str) -> String {
    format!("hello, {name}")
}

pub struct Widget {
    pub id: u32,
}

impl Widget {
    pub fn gamma_describe(&self) -> String {
        format!("widget-{}", self.id)
    }

    pub fn delta_clone(&self) -> Self {
        Widget { id: self.id }
    }
}

pub fn epsilon_dispatch(w: &Widget) -> String {
    w.gamma_describe()
}
